JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesRDBClient.java

```diff
@@ -101,31 +101,32 @@ protected void loadRdbConfigs(String filepath) throws IOException {
 			throw new JPSRuntimeException(exceptionPrefix + "No properties file found at specified filepath: " + filepath);
 		}
 		
-		// Open input stream from file
-		InputStream input = new FileInputStream(file);
-		
-		// Load properties file from specified path
-        Properties prop = new Properties();
-        prop.load(input);
-        // Close the input stream
-        input.close();
-
-        // Get the property values and assign
-        if (prop.containsKey("db.url")) {
-        	setRdbURL(prop.getProperty("db.url"));
-        } else {
-        	throw new JPSRuntimeException(exceptionPrefix + "Properties file is missing \"db.url=<rdb_url>\" ");
-        }
-        if (prop.containsKey("db.user")) {
-        	setRdbUser(prop.getProperty("db.user"));
-        } else {
-        	throw new JPSRuntimeException(exceptionPrefix + "Properties file is missing \"db.user=<rdb_username>\" ");
-        }
-        if (prop.containsKey("db.password")) {
-        	setRdbPassword(prop.getProperty("db.password"));
-        } else {
-        	throw new JPSRuntimeException(exceptionPrefix + "Properties file is missing \"db.password=<rdb_password>\" ");
-        }
+		// Try-with-resource to ensure closure of input stream
+		try (InputStream input = new FileInputStream(file)) {
+		
+			// Load properties file from specified path
+	        Properties prop = new Properties();
+	        prop.load(input);
+	
+	        // Get the property values and assign
+	        if (prop.containsKey("db.url")) {
+	        	setRdbURL(prop.getProperty("db.url"));
+	        } else {
+	        	throw new JPSRuntimeException(exceptionPrefix + "Properties file is missing \"db.url=<rdb_url>\" ");
+	        }
+	        if (prop.containsKey("db.user")) {
+	        	setRdbUser(prop.getProperty("db.user"));
+	        } else {
+	        	throw new JPSRuntimeException(exceptionPrefix + "Properties file is missing \"db.user=<rdb_username>\" ");
+	        }
+	        if (prop.containsKey("db.password")) {
+	        	setRdbPassword(prop.getProperty("db.password"));
+	        } else {
+	        	throw new JPSRuntimeException(exceptionPrefix + "Properties file is missing \"db.password=<rdb_password>\" ");
+	        }
+		} catch (JPSRuntimeException e) {
+			throw e;
+		}
```

Comment:
You don't need to include the `catch` block here as any `JPSRuntimeException` (and any other type of exception) will just be passed up to the calling method by default if it isn't caught. The try-with-resources form effectively adds an implicit `finally` block that satisfies the requirement that something must happen if an exception is thrown in the `try` block.

Reply:
Unnecessary catch blocks have been removed with commit 5d46a6e03662bb9f552db77dc6881d7270868f76